app/build.gradle

```diff
@@ -108,14 +109,25 @@ dependencies {
     compile 'com.github.bumptech.glide:glide:3.7.0'
     compile 'io.reactivex:rxandroid:1.1.0'
     compile 'io.reactivex:rxjava:1.1.4'
-    compile "com.jakewharton:butterknife:$BUTTERKNIFE_VERSION"
-    compile 'com.jakewharton.timber:timber:4.1.0'
     compile('com.crashlytics.sdk.android:crashlytics:2.5.2@aar') {
         transitive = true;
     }
 
+    compile 'com.jakewharton.timber:timber:4.1.0'
+    compile "com.jakewharton:butterknife:$BUTTERKNIFE_VERSION"
+    apt     "com.jakewharton:butterknife-compiler:$BUTTERKNIFE_VERSION"
+
+    compile "com.google.auto.value:auto-value:$AUTO_VALUE_VERSION"
```

Comment:
I don't think you should include AutoValue as compile dependency because it contains Guava and other stuff which leaks to your APK.
I am using `provided 'com.jakewharton.auto.value:auto-value-annotations:1.2-update1'` together with usual `apt 'com.google.auto.value:auto-value:$AUTO_VALUE_VERSION'` to setup AutoValue